pyuvdata/calfits.py

```diff
@@ -40,6 +40,26 @@ def write_calfits(self, filename, spoof_nonessential=False,
         if run_check:
             self.check(run_check_acceptability=run_check_acceptability)
 
+        if self.Nfreqs > 1:
+            freq_spacing = self.freq_array[0, 1:] - self.freq_array[0, :-1]
+            if not np.isclose(np.min(freq_spacing), np.max(freq_spacing),
+                              rtol=self._freq_array.tols[0], atol=self._freq_array.tols[1]):
+                raise ValueError('The frequencies are not evenly spaced (probably '
+                                 'because of a select operation). The calfits format '
+                                 'does not support unevenly spaced frequencies.')
+        else:
+            freq_spacing = self.channel_width  # or should this be the length of the frequency range?
```

Comment:
frequency_range is a parameter that gives the range of valid frequencies for this cal solution (this is particularly useful for delay type cal solutions where there isn't a frequency axis but you need to know what range of frequencies the solution applies to). The question here is if there's only one frequency channel after the select, should the delta between the channels (something that is explicitly defined in fits arrays) be equal to the channel width or the width of the frequency range? Since frequency range is already reported in another header item, I'd lean toward the channel_width, but I wanted to flag it for discussion.